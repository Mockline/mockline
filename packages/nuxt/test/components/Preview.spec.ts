import { describe, it, expect } from "vitest";
import type {TypeOf} from "zod";
// @ts-ignore
import { MPreview } from "#components"
import ComponentRender from "../component-render";

describe('Preview', () => {
  it.each([
    ['base case', {}],
    ['<Preview>Slot</Preview>']
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MPreview.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MPreview)
    expect(html).toMatchSnapshot()
  })
})
