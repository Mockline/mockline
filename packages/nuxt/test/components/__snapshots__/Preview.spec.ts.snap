// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Preview > renders <Preview>Slot</Preview> correctly 1`] = `"<preview>Slot</preview>"`;

exports[`Preview > renders base case correctly 1`] = `
"<div class="inline-block isolate">
  <div class="transition-transform duration-200 ease-out origin-top-left cursor-zoom-in" style="cursor: zoom-in;"></div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
